src/main/java/com/exadel/discount/exception/GlobalExceptionHandler.java

```diff
@@ -0,0 +1,23 @@
+package com.exadel.discount.exception;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.MethodArgumentNotValidException;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+
+@ControllerAdvice
+public class GlobalExceptionHandler {
+    @ExceptionHandler(NotFoundException.class)
+    public ResponseEntity<ExceptionDetails> handleException(NotFoundException notFoundException) {
+        ExceptionDetails exceptionDetails = new ExceptionDetails(notFoundException.getMessage());
+        return new ResponseEntity<>(exceptionDetails, HttpStatus.NOT_FOUND);
```

Comment:
1. you don't need response entity here
2. Set status code using annotation
3. Are you log somewhere that exception had happened?

Reply:
+